wkcuber/api/Dataset.py

```diff
@@ -72,40 +130,88 @@ def get_layer(self, layer_name) -> Layer:
             )
         return self.layers[layer_name]
 
-    def add_layer(self, layer_name, category, dtype=None, num_channels=None, **kwargs):
-        if dtype is None:
-            dtype = np.dtype("uint8")
+    def add_layer(
+        self,
+        layer_name,
+        category,
+        dtype_per_layer=None,
+        dtype_per_channel=None,
+        num_channels=None,
+        **kwargs,
+    ):
         if num_channels is None:
             num_channels = 1
 
-        # normalize the value of dtype in case the parameter was passed as a string
-        dtype = np.dtype(dtype)
+        if dtype_per_layer is not None and dtype_per_channel is not None:
+            raise AttributeError(
+                "Cannot add layer. Specifying both 'dtype_per_layer' and 'dtype_per_channel' is not allowed"
+            )
+        elif dtype_per_channel is not None:
+            try:
+                dtype_per_channel = np.dtype(dtype_per_channel)
+            except TypeError as e:
+                raise TypeError(
+                    "Cannot add layer. The specified 'dtype_per_channel' must be a valid dtype. "
+                    + str(e)
+                )
+            dtype_per_layer = dtype_per_channel_to_dtype_per_layer(
+                dtype_per_channel, num_channels
+            )
+        elif dtype_per_layer is not None:
+            if type(dtype_per_layer) is not str:
+                dtype_per_layer = str(np.dtype(dtype_per_layer))
```

Comment:
An alternative to these lines would be to wrap a try-catch block around it and simply try to convert `dtype_per_layer` to a valid dtype. This would make the assertion that there is a digit in the dtype redundant (see `convert_dtypes`) because `str(np.dtype("int"))` results in `int64`.
What do you prefer?

Reply:
> An alternative to these lines would be to wrap a try-catch block around it and simply try to convert `dtype_per_layer` to a valid dtype. This would make the assertion that there is a digit in the dtype redundant (see `convert_dtypes`) because `str(np.dtype("int"))` results in `int64`.

Sounds good!